docs/docs/08-guides/03-react-vite-websockets.md

```diff
@@ -0,0 +1,492 @@
+
+  - [VSCode](https://marketplace.visualstudio.com/items?itemName=Monada.vscode-wing) - Official extension
+  - [IntelliJ](https://plugins.jetbrains.com/plugin/22353-wing) - Community extension
+
+## Step 1 - Installation & Scaffolding
+In this step, we will be creating our project by setting up two folders:
+
+ - `backend` - The place for Wing's backend code
+ - `frontend` - The place for our React app code
+
+### Creating a React App with Vite
+
+1. Create the root project folder:
+ ```sh
+ mkdir ~/shared-counter
+ cd ~/shared-counter
+ ```
+2. Create a new React app using Vite under the frontend folder:
+ ```sh
+ npm create vite frontend -- --template react-ts
+ ```
+3. Run your React app and ensure it works:
+ ```sh
+ cd frontend
+ npm install
+ npm run dev
+ ```
+> The result should be a very simple webpage that runs locally and works without any backend (note the counter is not shared between tabs).
+
+4. Press Ctrl-C to return to the CLI prompt.
+
+### Creating a Wing backend
+
+1. Install Wing:
+ ```sh
+ npm install -g winglang
+ # Verify installation
+ wing -V 
+ ```
+2. Create a `backend` directory under the project root:
+ ```sh
+ mkdir ~/shared-counter/backend
+ cd ~/shared-counter/backend
+ ```
+3. Generate a new empty Wing project:
+ ```sh
+ wing new empty
+ ```
+> This will generate three files: `package.json`, `package-lock.json` and `main.w` file with a simple hello-world `Cloud.Function`
+4. Run this project inside Wing Simulator
+```sh
+wing run main.w
+```
+> The result should be a page shows a single `Cloud.Function` if you invoke it, it should show `hello, world` in the response section.
+5. Ctrl-C to go back to CLI prompt.
+
+## Step 2 - Hello `@winglibs/vite` 
+
+In the previous step, we used `npm run dev` to start the local web server. 
+In this step, we will install the `@winglibs/vite` package responsible for starting the dev server. 
+We will also pass information from the backend to the frontend.
+
+### Install and use `@winglibs/vite`
+
+1. Install `@winglibs/vite`:
+ ```sh
+ cd ~/shared-counter/backend
+ npm i -s @winglibs/vite
+ ```
+1. Bring and instantiate Vite in `backend/main.w`:
+ ```wing
+ bring vite;
+ 
+ new vite.Vite(
+   root:"../frontend"
+ );
+ ```
+3. Run this project inside the Wing Simulator:
+ ```sh
+ wing run main.w
+ ```
+> You should have 2 web pages open: the browser and the Wing Console, showing the simulator.
+
+### Sending data to your Vite app using `publicEnv`
+
+Now that we have our backend instantiate the Vite resource, 
+we would like to see how we can pass constant data from the backend to the frontend.
+
+1. Use `publicEnv` for the title in `backend/main.w:
+ ```wing
+ bring vite;
+ 
+ new vite.Vite(
+   root:"../frontend",
+   publicEnv: {
+     title: "Wing + Vite + React"
+   }
+ );
+ ```
+2. Notice that the React webpage now has `window.wing.env` containing this title. 
```

Comment:
it is written in a confusing way in my opinion:
```
2. Our frontend application now has a new `title` property in `window.wing.env` object. Open the browser's developer tools console and run: `console.log(window.wing.env.title);` to see it's value.
```

Reply:
moved to async discussion, resolving